### DISCUSSION TITLE
connect-PnPOnline

### DISCUSSION THREAD

**Discussion by moatome** [pnp/powershell#4316] [Q&A]
in short, i have script that:
1.connects to a sharepoint site
2. gets the items of a list that were created today
3. for each item create a folder in the library 
4. add a link to the path of the folder in one of the columns of that item in the list
5. post a message in a channel that the item has been created 

as you can see in the picture i tried the following connections but all of them are resulting in errors / arent compatible 
any advice on how i can connect the SP in a compatible way to my script.

![thumbnail_IMG_4528](https://github.com/user-attachments/assets/52a36088-57b1-4144-882e-9b24fcb4af63)

**Comment by ToddKlindt** [comment]
try this to connect:
`Connect-PnPOnline -Url https://contoso.sharepoint.com/ -ClientId $ClientID -Interactive`
-WebLogin only gets a SharePoint token, so it's very limited. -Interactive will give you a Graph token, which is what you need.
